spec/lib/stellar/sep10_spec.rb

```diff
@@ -0,0 +1,604 @@
+require "spec_helper"
+
+describe Stellar::SEP10 do
+
+  subject(:sep10) { Stellar::SEP10 }
+
+  let(:server) { Stellar::KeyPair.random }
+  let(:user) { Stellar::KeyPair.random }
+  let(:anchor) { "SDF" }
+  let(:timeout) { 600 }
+  let(:envelope) { Stellar::TransactionEnvelope.from_xdr(subject, "base64") }
+  let(:transaction) { envelope.tx }  
+
+  subject do
+    sep10.build_challenge_tx(server: server, client: user, anchor_name: anchor, timeout: timeout) 
+  end
+  
+  describe "#build_challenge_tx" do      
+    it "generates a valid SEP10 challenge" do
+      expect(transaction.seq_num).to eql(0)
+      expect(transaction.operations.size).to eql(1);
+      expect(transaction.source_account).to eql(server.public_key);
+
+      time_bounds = transaction.time_bounds
+      expect(time_bounds.max_time - time_bounds.min_time).to eql(600)
+      operation = transaction.operations.first
+
+      expect(operation.body.arm).to eql(:manage_data_op)
+      expect(operation.body.value.data_name).to eql("SDF auth")
+      expect(operation.source_account).to eql(user.public_key)
+      data_value = operation.body.value.data_value
+      expect(data_value.bytes.size).to eql(64)
+      expect(data_value.unpack("m")[0].size).to eql(48)
+    end
+
+    describe "defaults" do
+      subject do
+        sep10.build_challenge_tx(server: server, client: user, anchor_name: anchor) 
+      end
+      
+      it "has a default timeout of 300 seconds (5 minutes)" do
+        time_bounds = transaction.time_bounds
+        expect(time_bounds.max_time - time_bounds.min_time).to eql(300)
+      end
+    end
+  end
+
+  describe "#read_challenge_tx" do
+    subject do
+      challenge = super()
+      envelope = Stellar::TransactionEnvelope.from_xdr(challenge, 'base64')
+      envelope.tx.to_envelope(server, user).to_xdr(:base64)
+    end
+    
+    it "returns the envelope and client public key if the transaction is valid" do
+      expect(sep10.read_challenge_tx(challenge: subject, server: server)).to eql([envelope, user.address])
+    end
+
+    it "throws an error if transaction sequence number is different to zero"  do
+      envelope.tx.seq_num = 1
+
+      expect { 
+        sep10.read_challenge_tx(challenge: envelope.to_xdr(:base64), server: server)
+      }.to raise_error(Stellar::InvalidSep10ChallengeError, /The transaction sequence number should be zero/)
+    end      
+
+    it "throws an error if transaction source account is different to server account id"  do
+      expect {
+        sep10.read_challenge_tx(challenge: envelope.to_xdr(:base64), server: Stellar::KeyPair.random)
+      }.to raise_error(Stellar::InvalidSep10ChallengeError, /The transaction source account is not equal to the server's account/)
+    end
+
+    it "throws an error if transaction doestn\'t contain any operation" do
```

Comment:
typo: `doestn\'t`

also this escaping isn't necessary, it can just be:  `"throws an error if transaction doesn't contain any operation"`